### PR TITLE
Update to latest D1 API

### PR DESCRIPTION
**Pull request by tom-sherman**
- [x] I have verified my changes didn't break the `example` project.

#### Description

D1 API has changed: https://github.com/cloudflare/workerd/blob/42d45ba4cd41ed714b737a605a1022cc8d24cdde/types/defines/d1.d.ts#L1-L6

~I have no idea if this is a bug or not lol~

~I also can't get this to work on the example project (can't get it to return the new result shape), which is leading me to think this is some sort of bug???? I am seeing the new shape in my Pages builds tho...~

~I'll leave this in draft for now while I hear wait to hear something in the cloudflare discord.~

I had the wrong compat date and wrangler version lol

### CODE REVIEW

**Comment by tom-sherman** [comment]
What really confused me is that there is a bug in `wrangler pages dev` right now that causes the old result shape to be returned locally but not when deployed. Raised that here: https://github.com/cloudflare/wrangler2/issues/2673

**Comment by aidenwallis** [comment]
I'm not sure, so is this broken or not? 👀 It sounds like you fixed the issue?

**Comment by tom-sherman** [comment]
@aidenwallis haha the branch is now correct and is working, it targets the latest D1 API. Need to merge this PR to continue using D1 with Kysely 😄

**Comment by aidenwallis** [comment]
Ah got it, thanks

File: example/src/index.ts [hunk 1]
```diff
@@ -50,11 +50,17 @@ export default {
...
+          await db
+            .insertInto('kv')
+            .values([{ key, value }])
+            .onConflict((oc) => oc.column('key').doUpdateSet({ value }))
+            .execute();
+        } catch (err) {
+          console.log(err);
+          console.log((err as any).cause);
+          throw err;
+        }
```
Comment (aidenwallis) [inline example/src/index.ts:63]: Is this meant to be here?

**Review by aidenwallis — COMMENTED** [review]
just one question about the example change

**Comment by aidenwallis** [comment]
Also, this will be a major release (0.2.0), given we're changing the meta object structure changes and we don't provide fallbacks for the old version.

File: example/src/index.ts (hunk 1 above)
Comment (tom-sherman) [reply example/src/index.ts:63]: I think it's good to add for debugging the example, without logging error.cause D1 errors are pretty useless.

**Comment by tom-sherman** [comment]
Sounds ok, I don't think there's any point maintaining backwards compatibility while D1 is in alpha.

**Review by aidenwallis — APPROVED** [review]

**Comment by aidenwallis** [comment]
Cut the release on NPM and Github: https://github.com/aidenwallis/kysely-d1/releases/tag/0.2.0

**Pull request merged**